### PR TITLE
Fix pi05 expert only speed

### PR DESCRIPTION
**Pull request by WeikeShi0730**
## Title

  fix(pi05): speed up expert-only training

  ## Summary / Motivation

  This PR fixes a PI05 expert-only training slowdown introduced by keeping the frozen PaliGemma prefix path on the expensive training path. When train_expert_only=true, PaliGemma is frozen, but the previous implementation still checkpointed frozen image/language prefix work and kept the full frozen vision tower/projector in fp32. This change detaches the frozen prefix branch, avoids unnecessary checkpointing for tensors without gradients, and lets the frozen heavy vision path run mostly in bf16 while preserving fp32 for sensitive embedding/norm parameters. Full-model training keeps the existing full-vision fp32 behavior.

  ## Related issues

  - Fixes / Closes: N/A
  - Related: N/A

  ## What changed

  - Detach the frozen PaliGemma prefix branch when train_expert_only=true while keeping the action expert branch trainable.
  - Avoid wrapping the already layer-checkpointed PI05 joint forward in an extra outer checkpoint.
  - Skip checkpointing when no tensor input requires gradients.
  - In expert-only mode, keep only sensitive vision embedding/norm parameters in fp32 and run the frozen vision tower/projector mostly in bf16.
  - No breaking changes expected.

  ## How was this tested (or how to run locally)

  - Tests added: none.
  - Local checks:
python -m py_compile src/lerobot/policies/pi05/modeling_pi05.py
git diff --check -- src/lerobot/policies/pi05/modeling_pi05.py

  - Manual CUDA timing probe with PI05, train_expert_only=true, dtype=bfloat16, gradient_checkpointing=true, batch size 16:
      - Before final dtype fix: steady updt_s around 6.6-7.8s.
      - After fix: steady updt_s around 1.67s after warmup.
      - 8-step probe full tqdm average: 3.14s/step including first-step data/startup overhead.

  Example probe shape:

  lerobot-train \
      --dataset.repo_id=your_dataset \
      --policy.type=pi05 \
      --output_dir=./outputs/pi05_training \
      --job_name=pi05_training \
      --policy.repo_id=your_repo_id \
      --policy.pretrained_path=lerobot/pi05_base \
      --policy.compile_model=false \
      --policy.gradient_checkpointing=true \
      --wandb.enable=false \
      --policy.dtype=bfloat16 \
      --policy.freeze_vision_encoder=false \
      --policy.train_expert_only=**true** \
      --batch_size=16

  ## Checklist (required before merge)

  - [x] Linting/formatting run (pre-commit run -a)
  - [ ] All tests pass locally (pytest)
  - [ ] Documentation updated
  - [ ] CI is green
  - [ ] Community Review: I have reviewed another contributor's open PR and linked it here: # (insert PR number/link)

  ## Reviewer notes

  Please focus on whether the expert-only branch correctly prevents gradients through the frozen PaliGemma prefix while preserving gradients through the action expert. The dtype change is intentionally conditional: full-model fine-tuning keeps the existing fp32 vision path, while expert-only fine-tuning uses bf16 for the frozen heavy vision path.

### CODE REVIEW

**Comment by WeikeShi0730** [comment]
Hi @pkooij, could you please take a look at this PR when you have bandwidth?
The change is scoped to PI05 expert-only training. The main thing I’d like review on is whether detaching the frozen PaliGemma prefix path is the right approach while preserving gradients through the action expert branch.
